### DISCUSSION TITLE
Multiple Implementations of JoltC

### DISCUSSION THREAD

**Discussion by a-day-old-bagel** [amerkoleci/JoltPhysicsSharp#25] [Announcements]
I noticed that both the C# port _as well as_ the Zig port of Jolt implement a C port of Jolt underneath to use as a compatibility layer (both happen to be named JoltC, even).

Is it an issue that there are two different C APIs reproducing the same work? Could it benefit from more collaboration?

**Comment by amerkoleci** [comment]
C# and Zig are quite different, and the C bindings I've written are different and oriented to be used from C# world, for example the callback logic (https://github.com/amerkoleci/JoltPhysicsSharp/blob/main/src/joltc/joltc.h#L551)

The fact that they have same name doesn't mean they are immediatelly compatible.

**Comment by a-day-old-bagel** [comment]
Well I only ask because I think that's actually fairly generalizable. I imagine that a C API should function as a C API first, and a compatibility layer as a side effect of that. I was reading over the two different C API implementations and remarking how similar they were, which isn't surprising, since they're going to converge on the same interface naturally. The differences between Zig and C# would ideally make no difference to the C API.

But if nobody else thinks it's worth worrying about, then I won't worry either. Just wanted to ask. Thanks for the quick reply.
